### ISSUE TITLE
El Capitan issues

### ISSUE THREAD

**Issue by Chriz007** [w0lfschild/cDock#34]
It just won't work at all. 

Re-installed cdock, disabled rootless, but it still doesn't work.

**Comment by herbs** [comment]
Howdy,

I assume you have to first disable rootless, re-boot, re-install cdock (actually SIMBL is what needs to be installed using rootless). You can then re-enable rootless and re-boot.

Good Luck,
Herb Schulz

**Comment by w0lfschild** [comment]
That is correct herbs.

Also it should be noted that nvram commands do not disable rootless anymore. You must boot to recovery and run the security tool to toggle rootless on or off.

**Comment by Chriz007** [comment]
Thank you guys! All I had to do was disable the integrity protection during the recovery mode. You can close this issue now

**Comment by Chriz007** [comment]
Also, any  plans to have more themes added?

**Comment by w0lfschild** [comment]
No not really. If you want a more personalized theme it's best to explore how to make your own custom theme, there is documentation for every theme option. I am open to adding themes if other people make them like the Lion theme I just won't add any more themes with images unless the image files are very small in terms of disk space.

This can stay open for the moment as I suspect many users will not understand how to turn off rootless (or that rootless is even a thing) when El Capitan is actually released. If I do not create a warning window with instructions on how to disable rootless before 10.11 is released I suspect I'll be getting plenty of emails from users complaining that cDock won't work.

**Comment by w0lfschild** [comment]
Note with the latest recovery update whether intentional or not the configuration utility has been removed.

You must now:
- boot to recovery
- open terminal
- enter    
     `csrutil disable`
- reboot

**Comment by copykatz** [comment]
Evening—

Just installed El Capitan, then using the instructions, disabled system integrity protection, restarted and tried CDock.  No error message like I got when E.Cap was 1st launched, so thought all was well.  CDock panel comes up, but the dock remained grey instead of going to my choice of transparent, even tho the dock blanks out as if it were resetting.  Hit Apply several times, no change.  Tried uninstalling (using appzapper) cDock and then reinstalled.  No change.  cDock flashes the dock, but transparancy isn’t there.  I haven’t yet re-enabled SIP but will go ahead and do so—I was only going to do that once cDock gave me the transparent dock.  Any suggestions on what to try?

**Comment by herbs** [comment]
> On Sep 30, 2015, at 6:40 PM, copykatz <notifications@github.com> wrote:
> 
> Evening—
> 
> Just installed El Capitan, then using the instructions, disabled system integrity protection, restarted and tried CDock. No error message like I got when E.Cap was 1st launched, so thought all was well. CDock panel comes up, but the dock remained grey instead of going to my choice of transparent, even tho the dock blanks out as if it were resetting. Hit Apply several times, no change. Tried uninstalling (using appzapper) cDock and then reinstalled. No change. cDock flashes the dock, but transparancy isn’t there. I haven’t yet re-enabled SIP but will go ahead and do so—I was only going to do that once cDock gave me the transparent dock. Any suggestions on what to try?
> 
> —
> Reply to this email directly or view it on GitHub.
> 

Howdy,

No problem here. I disabled SIP/Rootless using the method suggested at the Bartender site (another great piece of software), did a normal reboot and login (was asked for my admin password by cDock (I assume cDockHelper in reality) as well as Bartender to install software and then enabled SIP/Rootless and all is well. (Well there is a small problem with Bartender 2 at login, but that's another story and not relevant here.

Good Luck,

Herb Schulz
(herbs at wideopenwest dot com)

**Comment by copykatz** [comment]
I’m glad it worked for you and you had no problem—doing the exact same thing you did, still gives me the grey background.  I therefore still need the suggestions for help that I asked for earlier to get this working.

**Comment by OSXMHH** [comment]
Downloaded 10.11 just today, noticed that clock wasn't fully functional followed the utility usable process and now it seems to work. I have preferred the older layout and style of both the dock & folders (the recent OS X releases are to harsh and tire the eyes) anyway The dock took the transparency of the Lion theme but it dock is still 90° to the view instead of at an oblique or perspective view. Is that not possible any more?
Thanks

**Comment by w0lfschild** [comment]
@copykatz you need to make sure you followed the proper procedure to disable SIP then with it disabled you need to run cDock which should prompt you to install SIMBL.

**Comment by copykatz** [comment]
That I did--abut 6 or 7 times.  :)

**Comment by bhaase** [comment]
Same here as copykatz.  Disabled SIP and installed Cdock then enabled SIP.  SIMBL installs properly and shows up in library/applicationsupport directory.  Dock stays grey, wont switch to transparent or any other background.  Have tried many times, same results. Any ideas?  This was an El Capitan upgrade, not a fresh install.  Even removed all references to dockmod and previous SIMBL installations.

**Comment by w0lfschild** [comment]
SIMBL installs @ /System/Library/ScriptingAdditions/SIMBL.osax not /Library/Application\ Support

**Comment by bhaase** [comment]
Yes its there. But not working? Does that need to be deleted to try again? No matter what i try it doesnt work

**Comment by bhaase** [comment]
Tried latest release as well..no luck.

**Comment by grkka** [comment]
Hello, I just updated c.Dock and El Capitan, not does the transparent theme. I've done these steps as indicated, but not working.
boot to recovery
open terminal
enter
csrutil disable
reboot

**Comment by bhaase** [comment]
Yes. Have gone through those steps. Not working for me. Thanks anyway.

**Comment by herbs** [comment]
Howdy,

Please run

ls -l /System/Library/ScriptingAdditions/

in Terminal. The listing should contain a line similar to

drwxr-xr-x  3 root   wheel  102 Sep 30 15:42 SIMBL.osax

and it it doesn't something went wrong when you disabled SIP and re-booted normally and ran cDock.

Good Luck,

Herb Schulz
(herbs at wideopenwest dot com)

**Comment by bhaase** [comment]
Yes. I have that line in terminal shown correctly.

**Comment by ferences** [comment]
Hello Wolfgang,

Thank you so much(with my donation).
My beautiful transparent dock(cDock 2) back in El Capitan.

Please keep up your great work.
Ference

**Comment by rmvz3** [comment]
Same here. Everything done. SIMBL is installed. No Theme in my dock. :(

**Comment by copykatz** [comment]
I keep trying Dock2, also--everything blanks out, dashboard resets, dock
goes away and when it comes back, nothing.  I was looking forward to
setting the application dots to another color, too.  :)

**Comment by bhaase** [comment]
I finally got things working. I deleted users/library/preferences/com.apple.dock.plist and com.apple.dock.extras.plist, rebooted and re-installed all while having rootless disabled.  Seems to have done the trick.  I think there were some leftovers from Yosemite causing a problem.

**Comment by copykatz** [comment]
That helped--sort of. Deleted the 2 pref files as you suggested.  cDock now
shows me a transparent background so I'm at least that far.

However, cDock2 still doesn't apply anything--grey-bars the dock's
background, causes the dock to quit and restart, but no changes that I can
see.

At least I'm back where I was pre-E.C.   It's a start.  :)

**Comment by copykatz** [comment]
Hmmmm....this is interesting.

As reported, cDock2 doesn't give me a transparent dock--my very preferred
dock theme (and continued thanks for doing that, Wolf!)

Since cD2 didn't give me transparency, I went back and ran cDock again.
When the transparent dock came back, the application indicator dots were
the color I had chosen in cD2.

Ran the sequence again--cD2 never does give me a transparent dock, but
quitting that and running cDock(1) gives me transparent and colored
indicators.  Settings both for dock transparency and colored indicators are
there after restarts.

So--multistep process, but it looks like running both programs gives me the
settings I like.

Howard

**Comment by w0lfschild** [comment]
Try 0.9.8

**Comment by herbs** [comment]
> On Oct 10, 2015, at 10:33 PM, Wolfgang Baird <notifications@github.com> wrote:
> 
> Try 0.9.8
> 
> —
> Reply to this email directly or view it on GitHub.
> 

Howdy,

0.9.8 seems to run find here. But 0.9.7 was also working fine on my system.

Good Luck,

Herb Schulz
(herbs at wideopenwest dot com)

**Comment by copykatz** [comment]
Sorry, but no change here--cd2 doesn't change the dock from gray to
anything else.  I still ahve to close it down, run cD1 to get back to
transparent dock, and the colored indicators then are there.  Same as the
last version of cD2

**Comment by brandonmhp** [comment]
Does the "Folders on Top" feature work in 10.11? Wanted to check before I jump through the SIP hoops.

**Comment by blatantfool** [comment]
I was having an issue with the dock theming not working, checked the console log and found that the cDock.bundle wasn't loading. When I followed it up, I found:

           /Library/Application Support/SIMBL/Plugins/cDock.bundle -> /Applications/cDock.app/Contents/Resources/bundles/cDock.bundle

but there is no bundles directory in the Resources directory. Changed the simlink to point to 

           /Applications/cDock.app/Contents/Resources/bundles/cDock.bundle

and now everything works. (OS X 10.11, cDock 2 version 0.9.8 build 1297)

**Comment by copykatz** [comment]
Could you give a step-by-step of how you changed it?  Not all of us are
that conversant.

**Comment by blatantfool** [comment]
This assumes you have admin privileges. Open Terminal.app .
```
 sudo bash
     <enter your admin password>
 cd /Library/Application\ Support/SIMBL/Plugins/
 ls -lh
```
If you have the problem I described, you will see a link between `cDock.bundle` and a non-existent file in  `/Applications/cDock.app/Contents/Resources/bundles/`  .
```
 ls /Applications/cDock.app/Contents/Resources/
```
This will show that there is no `bundles` directory
```
 ls -ld /Applications/cDock.app/Contents/Resources/cDock.bundle/
```
This will show that the `cDock.bundle` is now directly in resources. Assuming everything is as described:
```
 mv cDock.bundle cDock.bundle.dist
 ln -s cDock.bundle /Applications/cDock.app/Contents/Resources/cDock.bundle
```
This will create the necessary simlink to the right bundle. There’s really no benefit in moving the original simlink (the `mv` command), except to remind you later that you did something non-standard.

If you don't want the clutter lying around, you can do:
```
  rm cDock.bundle
```
instead of the `mv`

**Comment by herbs** [comment]
> On Oct 16, 2015, at 11:49 AM, blatantfool <notifications@github.com> wrote:
> ...
> If you have the problem I described, you will see a link between cDock.bundle and a non-existent file in /Applications/cDock.app/Contents/Resources/bundles/ .
> ...

Howdy,

On my system I actually do have a directory called cDock.bundle in /Library/Application\ Support/SIMBL/Plugins/.

Oh... cDock has been working fine on my system.

Good Luck,

Herb Schulz
(herbs at wideopenwest dot com)

**Comment by copykatz** [comment]
Ok, tried this using the terminal commands.

Did nothing. After doing all of that, launched cD2 (0.9.7) and it never
changed the dock from gray to transparent, no matter what I tried.  Closed
the program, ran as I did before, cD1 and I again get transparent dock with
the colored indicator dots.  So, while I appreciate your listing the steps
for me, it didn't let me use cD2 to change the dock--still need the last
cDock to make the changes visible!

**Comment by Custos** [comment]
Just wanted to highlight the same issue w/ Cdock2, installed through rootless, SIMBL appears in /system/library.  

drwxr-xr-x@ 3 root  wheel  102 Jan 12 13:38 SIMBL.osax

cDock2 doesn't do anything other than restart the dock.

**Comment by Custos** [comment]
Killed the dock and it appears everything is working now.

**Comment by copykatz** [comment]
I tried that too--still the same for me--two-step launch and need the old
cdock to get the colored dots to work--but I can live with that to get the
transparent dock w/ yellow dots  (mostly black background)  Glad yu got it
working for you, tho, Josh.

**Comment by DarkAthena** [comment]
I was having the same issues at the folks above. I disabled SIP and installed cDock as instructed. No dice. The dock wouldn't change. Opened console and saw that I was getting this error:

1/17/16 11:04:11.280 AM cDock 2[2357]: Cannot make directory /Library/Application Support/SIMBL/Plugins/cDock.bundle: Permission denied

So...I opened Macintosh HD/Library/Application Support/SIMBL/Plugins and saw that that directory, "Plugins" was set as "Read Only" for "everyone". Okay, we can fix that.

I selected "everyone" and clicked the padlock icon, entered my admin credentials in the authentication box that came up and set "everyone" as "Read & Write". 

Relaunched cDock and set my theme and it worked. Once cDock was able to write cDock.bundle into that folder, everything worked. I reset "everyone" as "Read Only" and my settings stuck.

I hope this helps someone else. Good luck!
